<!-- 
This launch file starts soundplay_node.py, which is the node that plays
sounds based on messages on the robotsound topic.  
-->

<launch>
  <arg name="cache_saying_file" default="false"
       doc="enables caching wave file automatically generated by festival" />
  <arg name="sound_file_prefix" default="sound_play"
       doc="prefix of generated wave / text file in temporary directory for saying" />
  <arg name="cache_timeout" default="120.0"
       doc="sound objects which are inactive more than cache_timeout will be removed" />
  <arg name="update_rate" default="5.0"
       doc="rate for polling audio play state" />

  <node name="soundplay_node" pkg="sound_play" type="soundplay_node.py">
    <rosparam subst_value="true">
      cache_saying_file: $(arg cache_saying_file)
      sound_file_prefix: $(arg sound_file_prefix)
      cache_timeout: $(arg cache_timeout)
      update_rate: $(arg update_rate)
    </rosparam>
  </node>
</launch>
